lib/core/form/components/widgets/core/form-field.model.ts

```diff
@@ -293,17 +293,14 @@ export class FormFieldModel extends FormWidgetModel {
         if (json.type === FormFieldTypes.DROPDOWN) {
 
             if (json.options) {
-                const options = <FormFieldOption[]> json.options || [];
-                if (options.length > 0) {
-                    if (json.hasEmptyValue) {
-                        const emptyOption = json.options[0];
-                        if (value === '' || value === emptyOption.id || value === emptyOption.name) {
-                            value = emptyOption.id;
-                        }
-                    } else {
-                        if (value?.id && value?.name) {
-                            value = value.id;
-                        }
+                if (json.hasEmptyValue) {
+                    const emptyOption = json.options[0];
+                    if (value === '' || value === emptyOption.id || value === emptyOption.name) {
```

Comment:
We need to unify and simplify this...
value should never be === to name (label)